google/cloud/sql/connector/InstanceConnectionManager.py

```diff
@@ -29,21 +29,27 @@
 import json
 import OpenSSL
 import socket
+from tempfile import NamedTemporaryFile
 import threading
 from typing import Any, Dict, Union
 
 import logging
 
 logger = logging.getLogger(name=__name__)
 
+  
 
 class InstanceMetadata:
+    ephemeral_cert: str
```

Comment:
These should probably have filepath in the name, so it is clear they are files? 

You probably also want to store a reference to the FileObject itself in this class, so that it doesn't get garbage collected before it's supposed to.

Reply:
The `InstanceMetadata` object is created in `_get_instance_data`, I can just move the creation of the temporary files into `_get_instance_data` or even into the constructor of the `InstanceMetadata` object.